File: library/src/main/java/cz/developer/library/ui/operation/DebugOperationFragment.java
package cz.developer.library.ui.operation;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.quant.titlebar.TitleBarFragment;

import java.io.IOException;

import cz.developer.library.R;

/**
 * Created by cz on 15/11/30.
 * debug其他设置界面
 */
public class DebugOperationFragment extends TitleBarFragment {
    @Nullable
    @Override
    public View onCreateView(Context context, LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_debug_other, container, false);
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        view.findViewById(R.id.tv_clear_cache).setOnClickListener(v -> clearAppData());
    }

    @Override
    public void onActivityCreated(Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        setTitleText(R.string.debug_info);
        setOnBackClickListener(v->getFragmentManager().popBackStack());
    }

    private void clearAppData() {
        String appPackage = getContext().getPackageName();
        String cmd = "pm clear " + appPackage + " HERE";
        Runtime runtime = Runtime.getRuntime();
        try {
            runtime.exec(cmd);
        } catch (IOException e) {
        }
    }
}

File: library/src/main/java/cz/developer/library/ui/operation/DebugOperationFragment.kt
package cz.developer.library.ui.operation

import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.app.AppCompatActivity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import java.io.IOException

import cz.developer.library.R
import kotlinx.android.synthetic.main.fragment_debug_other.*

/**
 * Created by cz on 15/11/30.
 * debug其他设置界面
 */
class DebugOperationFragment : Fragment() {
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_debug_other, container, false)
    }

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        clearCache.setOnClickListener { _ ->
            val appPackage = context.packageName
            val cmd = "pm clear $appPackage HERE"
            val runtime = Runtime.getRuntime()
            try {
                runtime.exec(cmd)
            } catch (e: IOException) {
            }
        }
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        val activity=activity
        if(activity is AppCompatActivity){
            toolBar.title = arguments?.getString("title")
            toolBar.subtitle=arguments?.getString("desc")
            activity.setSupportActionBar(toolBar)
            activity.supportActionBar?.setDisplayHomeAsUpEnabled(true)
            toolBar.setNavigationOnClickListener{ fragmentManager.popBackStack() }
        }
    }
}
